service/controller/resource/tcnp/create.go

```diff
@@ -350,10 +354,6 @@ func newLaunchConfiguration(ctx context.Context, cr infrastructurev1alpha2.AWSMa
 	return launchConfiguration, nil
 }
 
-func newLifecycleHooks(ctx context.Context, cr infrastructurev1alpha2.AWSMachineDeployment) (*template.ParamsMainLifecycleHooks, error) {
-	return &template.ParamsMainLifecycleHooks{}, nil
-}
```

Comment:
The separate lifecycle hook template got dropped because the hook moved into the ASG template. Defining the lifecycle hook inline the ASG fixes the problem of Node Pools not being drained during deletion.